components/notifier/src/quality_time_notifier.py

```diff
@@ -37,13 +34,11 @@ async def notify(log_level: int = None) -> NoReturn:
         except Exception as reason:  # pylint: disable=broad-except
             logging.error("Could not get reports from %s: %s", reports_url, reason)
             json = dict(reports=[])
-
-        for notification in get_notable_metrics_from_json(data_model, json, most_recent_measurement_seen):
-            for configuration in cast(Dict, notification["notification_destinations"]).values():
-                if configuration["teams_webhook"]:
-                    send_notification_to_teams(
-                        str(configuration["teams_webhook"]), build_notification_text(notification))
-
+        notifications = get_notable_metrics_from_json(data_model, json, most_recent_measurement_seen)
+        if len(notifications) > 0:
+            ready_to_send_metrics = outbox.outbox(notifications, ready_to_send_metrics)
+        else:
+            ready_to_send_metrics = notifications
```

Comment:
Is this code correct? If there are notifications in the ready_to_send_metrics list (=the outbox) and there are no new notifications, will ready_to_send_metrics be cleared without the notifications in ready_to_send_metrics being sent?

```suggestion
        ready_to_send_metrics = outbox.process_outbox(ready_to_send_metrics, notifications)
```

Reply:
Accepted and written a new test for empty lists.

Reply:
The code is still incorrect IMO.